src/sinks/util/encoding.rs

```diff
@@ -0,0 +1,89 @@
+use crate::event::{self, Event};
+
+use bytes::Bytes;
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug, Eq, PartialEq, Clone)]
+#[serde(rename_all = "snake_case")]
+pub enum BasicEncoding {
```

Comment:
I'm not sure if this is the right approach. We originally decided against centralizing it like this as some sinks use different types and the concept of encoding is per sink. I think we should move this back to each sink and possibly open an issue and another PR making this change.